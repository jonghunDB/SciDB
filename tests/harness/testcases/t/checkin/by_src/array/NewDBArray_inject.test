--setup
--start-query-logging

# Inject Error Support
load_library('misc')

--test
--error "store(apply(build(<val:int64>[i=0:100:0:10], i), inject, injectError(0,12)), chunkOutOfBoundsArray)"

--start-igdata
store(build(<val:int64>[i=0:100:0:10], i), noCurrentChunkArray)
--stop-igdata
--error "apply(scan(noCurrentChunkArray), inject, injectError(0,13))"

--error "apply(scan(noCurrentChunkArray), inject, injectError(0,14))"

--cleanup
--justrun "remove(chunkOutOfBoundsArray)"
--justrun "remove(noCurrentChunkArray)"
--stop-query-logging
