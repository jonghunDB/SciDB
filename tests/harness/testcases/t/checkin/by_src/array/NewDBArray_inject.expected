SCIDB QUERY : <load_library('misc')>
Query was executed successfully

SCIDB QUERY : <store(apply(build(<val:int64>[i=0:100:0:10], i), inject, injectError(0,12)), chunkOutOfBoundsArray)>
[An error expected at this place for the query "store(apply(build(<val:int64>[i=0:100:0:10], i), inject, injectError(0,12)), chunkOutOfBoundsArray)". And it failed.]

SCIDB QUERY : <store(build(<val:int64>[i=0:100:0:10], i), noCurrentChunkArray)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <apply(scan(noCurrentChunkArray), inject, injectError(0,13))>
[An error expected at this place for the query "apply(scan(noCurrentChunkArray), inject, injectError(0,13))". And it failed.]

SCIDB QUERY : <apply(scan(noCurrentChunkArray), inject, injectError(0,14))>
[An error expected at this place for the query "apply(scan(noCurrentChunkArray), inject, injectError(0,14))". And it failed.]

SCIDB QUERY : <remove(chunkOutOfBoundsArray)>
[SciDB query execution failed. But continuing, as it was intended to just run.]

SCIDB QUERY : <remove(noCurrentChunkArray)>
[Query was executed successfully. It was only intended to just run.]
